### PR TITLE
refactor(plugin): enable plugin to run in different modes

### PR DESCRIPTION
**Pull request by peknur**
- enable running modes: `node`, `controller`, and `monolith` with different runtime requirements
- refactor package layout to support new modes and split code to smaller testable units

### CODE REVIEW

**Review by villevsv-upcloud — APPROVED** [review]
Seems to be logically arranged & refactored.

File: internal/identity/identity.go [hunk 1]
```diff
@@ -44,14 +55,13 @@ func (d *Driver) GetPluginCapabilities(ctx context.Context, req *csi.GetPluginCa
...
-	d.log.WithField("method", "probe").Info("check whether the plugin is ready")
-	d.readyMu.Lock()
-	defer d.readyMu.Unlock()
+func (i *Identity) Probe(ctx context.Context, req *csi.ProbeRequest) (*csi.ProbeResponse, error) {
+	i.log.WithField("method", "probe").Info("check whether the plugin is ready")
 
 	return &csi.ProbeResponse{
 		Ready: &wrappers.BoolValue{
-			Value: d.ready,
+			// TODO: should we check e.g. that network is available or that all the cli tools are available ?
```
Comment (ajmyyra) [inline internal/identity/identity.go:63]: CLI tools check makes sense atleast. Perhaps network as well.

**Review by ajmyyra — APPROVED** [review]
Good refactoring here! Some TODOs still, but it doesn't need to solve everything, so looking good. Left one comment if you feel like adding some dependency checks there.

File: .goreleaser.yaml [hunk 2]
```diff
@@ -4,7 +4,10 @@ before:
...
-    - ./cmd/upcloud-csi-manifest/upcloud-csi-manifest --crd=false --rbac=true --setup=true  --output=./temp/{{ .ProjectName }}-setup.yaml --driver-version={{ if .IsSnapshot }}main{{ else }}v{{ .Version }}{{ end }}
+    - ./cmd/upcloud-csi-manifest/upcloud-csi-manifest --crd=false --rbac=true --setup=true --zone=de-fra1 --output=./temp/{{ .ProjectName }}-setup-de-fra1.yaml --driver-version={{ if .IsSnapshot }}main{{ else }}v{{ .Version }}{{ end }}
+    - ./cmd/upcloud-csi-manifest/upcloud-csi-manifest --crd=false --rbac=true --setup=true --zone=fi-hel2 --output=./temp/{{ .ProjectName }}-setup-fi-hel2.yaml --driver-version={{ if .IsSnapshot }}main{{ else }}v{{ .Version }}{{ end }}
+    - ./cmd/upcloud-csi-manifest/upcloud-csi-manifest --crd=false --rbac=true --setup=true --zone=sg-sin1 --output=./temp/{{ .ProjectName }}-setup-sg-sin1.yaml --driver-version={{ if .IsSnapshot }}main{{ else }}v{{ .Version }}{{ end }}
+    - ./cmd/upcloud-csi-manifest/upcloud-csi-manifest --crd=false --rbac=true --setup=true --zone=uk-lon1 --output=./temp/{{ .ProjectName }}-setup-uk-lon1.yaml --driver-version={{ if .IsSnapshot }}main{{ else }}v{{ .Version }}{{ end }}
```
[the remark this answers is not in the document]
Comment (peknur) [reply .goreleaser.yaml:10]: @villevsv-upcloud thanks for the review :+1:  This sound good, I'll try to implement it and push changes.

File: internal/controller/controller.go [hunk 3]
```diff
@@ -27,9 +29,32 @@ var supportedCapabilities = []csi.ControllerServiceCapability_RPC_Type{ //nolint
 	csi.ControllerServiceCapability_RPC_CLONE_VOLUME,
 }
 
+type Controller struct {
```
[the remark this answers is not in the document]
Comment (peknur) [reply internal/controller/controller.go:32]: Good point, I'll check if this is something that could be done in this PR or maybe in separate PR.

File: internal/identity/identity.go (hunk 1 above)
Comment (peknur) [reply internal/identity/identity.go:63]: @ajmyyra thanks for the review :+1:  We have `checkSystemRequirements` function in the `internal/filesystem` tests that we could probably use here. I'll create separate PR for that.

**Pull request merged**